### DISCUSSION TITLE
CCPC2022-女生专场 SZTU_EndeavorGirls队总结

### DISCUSSION THREAD

**Discussion by zxr0000000** [CSGrandeur/s-1problem1day1ac#529] [Summary]
6题罚时高，银首。

开局看的C一道签到题。感觉是签到题，但是...我忘记弧度制和角度的转换了，然后就退出去看新题。

然后看到最多人过的A，记录下最大值，模拟最多走一轮，然后就开始写恶臭的代码了。最近一直陷入逃避现实，没怎么敲代码，最后用了deque 写了70多行，造了几个数据测了下之后，开局12分的时候过了。

然后发现C题别人哐哐过，想着，这道题不过，就完蛋了。然后就开始研究 1.570796 和 90 度之间的关系。在不断的测试中，经历了将近30分钟的斗争后，发现 1.570796 = pi / 2，然后突然惊醒，直接根据这个来分配周长就可以了。最后得出对应的公式 = 半径 * 角度。现在想来，那个时候，真是失了智了。因为忘记给出的角度可能大于180，WA了一发。

然后接下来，就开了当时过的人最多的G，一眼模拟，然后就开始敲。由于代码写的非常糟糕，在不断出造数据测试中，一共WA了两发之后，105分钟的时候过了。

然后开E，一眼模拟，开始敲。敲完之后，没看到歌曲是循环的，贡献了一发完全没有意义的WA，然后感觉可能要开两轮，感觉中间交界处可能会出现答案，并且根据一轮的改变值是否是负数来判断，然后WA了。发现忽略了改变值为0的情况。改了过程中把两轮去掉了，然后又WA一发，造案例发现，确实要两轮。最终WA3发169分钟过了。

当时榜上有两道题看起来可以开，分别是I和H，看了一眼I发现是图论，立马跑去H了。一眼字典树+dp。由于好久没写字典树了，立马掏出模板，开敲！199分钟一A了。

最后来到了I，想着虽然是图论，但是不过可能就要铜了，还是硬着头皮想吧。然后看错了题，以为是刷新的是点权，估计了一下时间复杂度，感觉会超时，但是没有想到别的方法了，就硬着头皮写了迪杰斯特拉。WA了两发之后，发现看错题了，然后改了下代码，交了一发TLE了。然后重读了一次题目，发现，直接用Bellman-Ford预处理，记录每个路径条数对应的最短路的长度，就可以了。WA3发，272分钟过了。

剩下28分钟，饥饿和疲劳瞬间一涌而上，想着acmer不能放弃，就继续看题。但是不出意外，果然看了F之后，没有一点想法，小想了一下之后，就去看J，题目没看完，比赛就结束了。

发现银首的时候，想到那极高的dirty率，觉得挺可惜的。

从过的六道题来看，这场比赛不算很难（当然可能是后面两道较为难的题刚好撞我知识树上了）。等题挂出来，再补下，后一道题目吧。然后极高的dirty律，还是UVA写少了。

**Comment by CSGrandeur** [comment]
# 还是UVA写少了😏
